types/react-select/src/Select.d.ts

```diff
@@ -202,16 +205,17 @@ export interface NamedProps<OptionType extends OptionTypeBase = { label: string;
   /* Select the currently focused option when the user presses tab */
   tabSelectsValue?: boolean;
   /* The value of the select; reflected by the selected option */
-  value?: ValueType<OptionType>;
+  value?: ValueType<OptionType, IsMulti>;
 
   defaultInputValue?: string;
   defaultMenuIsOpen?: boolean;
-  defaultValue?: ValueType<OptionType>;
+  defaultValue?: ValueType<OptionType, true | false>;
```

Comment:
It actually breaks one of the examples (OnSelectResetsInput) where `isMulti = true` but only a single object is provided.

Possibly could do another conditional statement here to say that we can expect both when `IsMulti extends true` but only a single object otherwise